### PR TITLE
refactor: Remove `forceIntersectBoundaries` from `Navigator`

### PR DESCRIPTION
**Pull request by andiwand**
This is an attempt to drop a `Navigator` hotfix from the past assuming that it has been fixed by one of the `Navigator` refactors.

blocked by
- https://github.com/acts-project/acts/pull/3437
- https://github.com/acts-project/acts/pull/3436

### CODE REVIEW

File: Core/include/Acts/Propagator/Navigator.hpp [hunk 1]
```diff
@@ -904,15 +897,10 @@ class Navigator {
       // Increase the index to the next one
       ++state.navigation.navBoundaryIndex;
     }
+
     // We have to leave the volume somehow, so try again
```
Comment (benjaminhuth) [inline Core/include/Acts/Propagator/Navigator.hpp:901]: Hmm that comment with "try again" seems to be outdated with this change, right?

**Review by benjaminhuth — APPROVED** [review]
If its green, it seems to work. Only the comments in the affected section may need to be rethought...

**Review by benjaminhuth — COMMENTED** [review]
A wait but there are some physmon differences...

**Comment by andiwand** [comment]
> A wait but there are some physmon differences...

I suspect that these are coming from the simulation and the low stats in ttbar. I can try to validate this with higher stats or look at a case where it changes

File: Core/include/Acts/Propagator/Navigator.hpp [hunk 2]
```diff
@@ -921,15 +914,10 @@ class Navigator {
...
     state.navigation.navBoundaries.clear();
     ACTS_VERBOSE(volInfo(state) << "Boundary navigation lost, re-targetting.");
```
Comment (noemina) [inline Core/include/Acts/Propagator/Navigator.hpp:920]: You can also get rid of this logging message I guess.

**Comment by andiwand** [comment]
@noemina I investigated the propagation errors a bit and I think I have a solution with the 3 PRs in combination
- https://github.com/acts-project/acts/pull/3436
- https://github.com/acts-project/acts/pull/3437

**Comment by andiwand** [comment]
closing in favor of https://github.com/acts-project/acts/pull/3437

**Pull request closed without merging**